examples_tests/0.ImportanceSamplingEnvMaps/main.cpp

```diff
@@ -0,0 +1,668 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#define _NBL_STATIC_LIB_
+#include <nabla.h>
+#include "../../../include/nbl/asset/filters/CSummedAreaTableImageFilter.h"
+#include "nbl/ext/FullScreenTriangle/FullScreenTriangle.h"
+#include "nbl/ext/ScreenShot/ScreenShot.h"
+#include "glm/glm.hpp"
+#include "../common/QToQuitEventReceiver.h"
+
+using namespace nbl;
+using namespace asset;
+using namespace core;
+using namespace video;
+
+using SATFilter = CSummedAreaTableImageFilter<false>;
+
+static core::smart_refctd_ptr<ICPUBuffer> computeLuminancePdf(smart_refctd_ptr<ICPUImage> envmap)
+{
+	const core::vector2d<uint32_t> envmapExtent = { envmap->getCreationParameters().extent.width, envmap->getCreationParameters().extent.height };
+	const uint32_t channelCount = getFormatChannelCount(envmap->getCreationParameters().format);
+	
+	const core::vector2d<uint32_t> pdfDomainExtent = { envmapExtent.X, envmapExtent.Y };
```

Comment:
`core::vector<>` is legacy (trying to get rid of it), just use a plain `std::array<uint32_t>`